src/net/metaplaylist/index.ts

```diff
@@ -0,0 +1,340 @@
+        return request({
+          url,
+          responseType: "text",
+          ignoreProgressEvents: true,
+        }).map(({ value }) => {
+          return {
+            type: "response" as "response",
+            value: {
+              responseData: value.responseData,
+            },
+          };
+        });
+      },
+
+      parser(
+        { response, url } : IManifestParserArguments<Document|string>
+      ) : IManifestParserObservable {
+        if (typeof response.responseData !== "string") {
+          throw new Error("Parser input must be string.");
+        }
+        const contents = parseMetaPlaylistData(response.responseData);
+        const manifestsInfos$ = contents.map((content) => {
+          const transport = transports[content.transport];
+          if (transport == null) {
+            throw new Error(
+              "Transport" + content.transport + "not supported in MetaPlaylist.");
+          }
+          const loader = transport.manifest.loader;
+
+          return loader({ url: content.url })
+            .filter((res): res is ILoaderResponse<Document> => res.type === "response")
+            .map((res) => {
+              return {
+                manifest: res.value.responseData,
+                url: content.url,
+                startTime: content.startTime,
+                endTime: content.endTime,
+                transport: content.transport,
+                textTracks: content.textTracks,
+              };
+            });
+        });
+
+        return Observable
+          .combineLatest(manifestsInfos$)
+          .mergeMap((combinedManifest) => {
+            const manifestsInfos =  combinedManifest;
+            const parsedManifestsInfo = manifestsInfos.map((manifestInfos) => {
+              const transport = transports[manifestInfos.transport];
+              const parser = transport.manifest.parser;
+              const fakeResponse = {
+                responseData: manifestInfos.manifest,
+              };
+              return parser({ response: fakeResponse, url: manifestInfos.url })
+              .map((mpd) => {
+                const type = mpd.manifest.type;
+                if (type !== "static") {
+                  throw new Error("Content from metaplaylist is not static.");
+                }
+                return {
+                  manifest: mpd.manifest,
+                  transport: manifestInfos.transport,
+                  url: manifestInfos.url,
+                  startTime: manifestInfos.startTime,
+                  endTime: manifestInfos.endTime,
+                  textTracks: manifestInfos.textTracks,
+                };
+              });
+            });
+            return Observable.combineLatest(parsedManifestsInfo).map((_contents) => {
+                const manifest = generateManifest(_contents, url);
+                return {
+                  manifest,
+                  url,
+                };
+            });
+          });
+      },
```

Comment:
This function is too complex, I couldn't see how I can integrate my overlay into that. Even with ``any`` everywhere and trying very hard.

This is too hard to read